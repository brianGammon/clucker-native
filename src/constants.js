export const APP_VERSION = '0.0.0-dev';
export const PRIVACY_POLICY_URI = 'https://cluckerapp.com';
export const TERMS_OF_USE_URI = 'https://cluckerapp.com';
export const BULK_ENTRY_KEY = 'bulkentry';
export const UNNAMED_HEN_KEY = 'unnamed';
export const UNNAMED_HEN_LABEL = 'Unnamed Hen';
