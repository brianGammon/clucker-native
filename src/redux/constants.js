// @flow
export const actionTypes = {
  CREATE_REQUESTED: 'CREATE_REQUESTED',
  CREATE_FULFILLED: 'CREATE_FULFILLED',
  CREATE_REJECTED: 'CREATE_REJECTED',

  UPDATE_REQUESTED: 'UPDATE_REQUESTED',
  UPDATE_FULFILLED: 'UPDATE_FULFILLED',
  UPDATE_REJECTED: 'UPDATE_REJECTED',

  REMOVE_REQUESTED: 'REMOVE_REQUESTED',
  REMOVE_FULFILLED: 'REMOVE_FULFILLED',
  REMOVE_REJECTED: 'REMOVE_REJECTED',

  LISTEN_REQUESTED: 'LISTEN_REQUESTED',
  LISTEN_FULFILLED: 'LISTEN_FULFILLED',
  LISTEN_REJECTED: 'LISTEN_REJECTED',

  LISTEN_REMOVED: 'LISTEN_REMOVED',

  REMOVE_LISTENER_REQUESTED: 'REMOVE_LISTENER_REQUESTED',
  REMOVE_ALL_LISTENERS_REQUESTED: 'REMOVE_ALL_LISTENERS_REQUESTED',

  CLEAR_ERROR: 'CLEAR_ERROR',
  CLEAR_AUTH_ERROR: 'CLEAR_AUTH_ERROR',

  SYNC_FLOCKS_REQUESTED: 'SYNC_FLOCKS_REQUESTED',
  SYNC_FLOCKS_REJECTED: 'SYNC_FLOCKS_REJECTED',
  SYNC_FLOCKS_FULFILLED: 'SYNC_FLOCKS_FULFILLED',

  SET_FLOCK: 'SET_FLOCK',
  CLEAR_FLOCK: 'CLEAR_FLOCK',
  CLEAR_ALL_FLOCKS: 'CLEAR_ALL_FLOCKS',

  UNLINK_FLOCK_REQUESTED: 'UNLINK_FLOCK_REQUESTED',
  UNLINK_FLOCK_FULFILLED: 'UNLINK_FLOCK_FULFILLED',

  DELETE_FLOCK_REQUESTED: 'DELETE_FLOCK_REQUESTED',
  DELETE_FLOCK_FULFILLED: 'DELETE_FLOCK_FULFILLED',
  DELETE_FLOCK_REJECTED: 'DELETE_FLOCK_REJECTED',

  DELETE_CHICKEN_REQUESTED: 'DELETE_CHICKEN_REQUESTED',
  DELETE_CHICKEN_FULFILLED: 'DELETE_CHICKEN_FULFILLED',
  DELETE_CHICKEN_REJECTED: 'DELETE_CHICKEN_REJECTED',

  JOIN_FLOCK_REQUESTED: 'JOIN_FLOCK_REQUESTED',
  JOIN_FLOCK_FULFILLED: 'JOIN_FLOCK_FULFILLED',
  JOIN_FLOCK_REJECTED: 'JOIN_FLOCK_REJECTED',

  ADD_FLOCK_REQUESTED: 'ADD_FLOCK_REQUESTED',
  ADD_FLOCK_FULFILLED: 'ADD_FLOCK_FULFILLED',
  ADD_FLOCK_REJECTED: 'ADD_FLOCK_REJECTED',

  SAVE_CHICKEN_REQUESTED: 'SAVE_CHICKEN_REQUESTED',

  CHILD_ADDED: 'CHILD_ADDED',
  CHILD_CHANGED: 'CHILD_CHANGED',
  CHILD_REMOVED: 'CHILD_REMOVED',

  SET_INITIAL_URL: 'SET_INITIAL_URL',
  REMOVE_INITIAL_URL: 'REMOVE_INITIAL_URL',

  AUTH_STATUS_LOGGED_IN: 'AUTH_STATUS_LOGGED_IN',
  AUTH_STATUS_LOGGED_OUT: 'AUTH_STATUS_LOGGED_OUT',

  SIGN_OUT_REQUESTED: 'SIGN_OUT_REQUESTED',

  AUTH_ACTION_REQUESTED: 'AUTH_ACTION_REQUESTED',
  AUTH_ACTION_FULFILLED: 'AUTH_ACTION_FULFILLED',
  AUTH_ACTION_REJECTED: 'AUTH_ACTION_REJECTED',

  STORAGE_DELETE_REQUESTED: 'STORAGE_DELETE_REQUESTED',
  STORAGE_DELETE_FULFILLED: 'STORAGE_DELETE_FULFILLED',
  STORAGE_DELETE_REJECTED: 'STORAGE_DELETE_REJECTED',

  STORAGE_UPLOAD_REQUESTED: 'STORAGE_UPLOAD_REQUESTED',
  STORAGE_UPLOAD_FULFILLED: 'STORAGE_UPLOAD_FULFILLED',
  STORAGE_UPLOAD_REJECTED: 'STORAGE_UPLOAD_REJECTED',
};

export const metaTypes = {
  userSettings: 'userSettings',
  flocks: 'flocks',
  chickens: 'chickens',
  eggs: 'eggs',
};

export const authTypes = {
  signUp: 'signUp',
  signIn: 'signIn',
  resetPassword: 'resetPassword',
};

export const eventTypes = {
  CHILD_ADDED_EVENT: 'CHILD_ADDED_EVENT',
  CHILD_REMOVED_EVENT: 'CHILD_REMOVED_EVENT',
  CHILD_CHANGED_EVENT: 'CHILD_CHANGED_EVENT',
  AUTH_STATUS_CHANGED: 'AUTH_STATUS',
};

export const appStates = {
  STARTING: 'STARTING',
  BUSY: 'BUSY',
  READY: 'READY',
};
