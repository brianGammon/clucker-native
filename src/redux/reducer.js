import {
  metaTypes, authTypes, actionTypes as a, appStates,
} from './constants';

const getAuthErrorsInitialState = () => {
  const errors = {};
  Object.keys(authTypes).forEach((key) => {
    errors[key] = null;
  });
  return errors;
};

const getInitialState = () => {
  const state = {
    appState: appStates.STARTING,
    initialUrl: null,
    auth: {
      inProgress: false,
      user: null,
    },
    joinForm: {
      inProgress: false,
      error: null,
    },
    addForm: {
      inProgress: false,
      error: null,
    },
    deleteFlock: {
      inProgress: false,
      error: null,
    },
    deleteChicken: {
      inProgress: false,
      error: null,
    },
  };
  Object.keys(metaTypes).forEach((key) => {
    const subState = {
      initialized: false,
      inProgress: false,
      error: null,
      data: {},
    };
    state[key] = subState;
  });
  state.auth.errors = getAuthErrorsInitialState();
  return state;
};

const initialState = getInitialState();

const handlers = {
  [a.AUTH_STATUS_LOGGED_IN](state, action) {
    const errors = getAuthErrorsInitialState();
    const newState = {
      ...state,
      appState: appStates.READY,
      auth: {
        inProgress: false,
        errors,
        user: action.payload,
      },
    };
    return newState;
  },
  [a.AUTH_STATUS_LOGGED_OUT](state) {
    const errors = getAuthErrorsInitialState();
    const newState = {
      ...state,
      appState: appStates.READY,
      auth: { inProgress: false, errors, user: null },
    };
    return newState;
  },
  [a.AUTH_ACTION_REQUESTED](state) {
    const errors = getAuthErrorsInitialState();
    const auth = { inProgress: true, errors, user: null };
    const newState = { ...state, auth };
    return newState;
  },
  [a.AUTH_ACTION_FULFILLED](state) {
    const errors = getAuthErrorsInitialState();
    const { user } = state.auth;
    const auth = { inProgress: false, errors, user };
    const newState = { ...state, auth };
    return newState;
  },
  [a.AUTH_ACTION_REJECTED](state, action) {
    const { error } = action.payload;
    const errorType = action.meta.type;
    const errors = { ...getAuthErrorsInitialState(), [errorType]: error.message };
    const auth = {
      inProgress: false,
      errors,
      user: null,
    };
    const newState = { ...state, auth };
    return newState;
  },
  [a.SET_INITIAL_URL](state, action) {
    const newState = { ...state, initialUrl: action.payload };
    return newState;
  },
  [a.REMOVE_INITIAL_URL](state) {
    const newState = { ...state, initialUrl: null };
    return newState;
  },
  [a.CLEAR_ERROR](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const newState = {
      ...state,
      [property]: { ...propertyState, error: null },
    };
    return newState;
  },
  [a.CLEAR_AUTH_ERROR](state) {
    const errors = getAuthErrorsInitialState();
    const newState = {
      ...state,
      auth: { ...state.auth, errors },
    };
    return newState;
  },
  [a.CREATE_REQUESTED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: true, error: null },
    };
    return newState;
  },
  [a.CREATE_FULFILLED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: false, error: null },
    };
    return newState;
  },
  [a.CREATE_REJECTED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const { error } = action.payload;
    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: false, error: error.message },
    };
    return newState;
  },
  [a.UPDATE_REQUESTED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: true, error: null },
    };
    return newState;
  },
  [a.UPDATE_FULFILLED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: false, error: null },
    };
    return newState;
  },
  [a.UPDATE_REJECTED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const { error } = action.payload;
    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: false, error: error.message },
    };
    return newState;
  },
  [a.REMOVE_REQUESTED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: true, error: null },
    };
    return newState;
  },
  [a.REMOVE_FULFILLED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: false, error: null },
    };
    return newState;
  },
  [a.REMOVE_REJECTED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const { error } = action.payload;
    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: false, error: error.message },
    };
    return newState;
  },
  [a.LISTEN_REQUESTED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];

    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: true, error: null },
    };
    return newState;
  },
  [a.LISTEN_FULFILLED](state, action) {
    const data = action.payload;
    const property = action.meta.type;
    const propertyState = state[property];

    const newState = {
      ...state,
      [property]: {
        ...propertyState,
        initialized: true,
        inProgress: false,
        error: null,
        data,
      },
    };
    return newState;
  },
  [a.LISTEN_REJECTED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const { error } = action.payload;

    const newState = {
      ...state,
      [property]: { ...propertyState, inProgress: false, error: error.message },
    };
    return newState;
  },
  // notice child added and changed are the same at the moment
  [a.CHILD_ADDED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const data = {
      ...propertyState.data,
      [action.payload.key]: action.payload.data,
    };

    const newState = {
      ...state,
      [property]: {
        ...propertyState,
        inProgress: false,
        error: null,
        data,
      },
    };
    return newState;
  },
  [a.CHILD_CHANGED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const data = {
      ...propertyState.data,
      [action.payload.key]: action.payload.data,
    };

    const newState = {
      ...state,
      [property]: {
        ...propertyState,
        inProgress: false,
        error: null,
        data,
      },
    };
    return newState;
  },
  [a.CHILD_REMOVED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const data = { ...propertyState.data };
    delete data[action.payload.key];

    const newState = {
      ...state,
      [property]: {
        ...propertyState,
        inProgress: false,
        error: null,
        data,
      },
    };
    return newState;
  },
  [a.LISTEN_REMOVED](state, action) {
    const property = action.meta.type;
    const propertyState = state[property];
    const data = action.payload.clearData ? {} : propertyState.data;

    const newState = {
      ...state,
      [property]: {
        ...propertyState,
        error: null,
        data,
      },
    };
    return newState;
  },
  [a.SYNC_FLOCKS_REQUESTED](state) {
    const propertyState = state.flocks;

    const newState = {
      ...state,
      flocks: {
        ...propertyState,
        inProgress: true,
        error: null,
      },
    };
    return newState;
  },
  [a.SYNC_FLOCKS_FULFILLED](state) {
    const propertyState = state.flocks;
    const newState = {
      ...state,
      flocks: {
        ...propertyState,
        initialized: true,
        inProgress: false,
      },
    };
    return newState;
  },
  [a.SYNC_FLOCKS_REJECTED](state, action) {
    const propertyState = state.flocks;
    const { error } = action.payload;

    const newState = {
      ...state,
      flocks: {
        ...propertyState,
        initialized: true,
        inProgress: false,
        error: error.message,
      },
    };
    return newState;
  },
  [a.SET_FLOCK](state, action) {
    const propertyState = state.flocks;
    const propertyData = propertyState.data;
    const newData = { ...propertyData, ...action.payload };

    const newState = {
      ...state,
      flocks: {
        ...propertyState,
        data: newData,
      },
    };
    return newState;
  },
  [a.CLEAR_FLOCK](state, action) {
    const { flocks } = state;
    const { [action.payload]: removed, ...rest } = flocks.data;
    const newState = { ...state, flocks: { ...flocks, data: rest } };
    return newState;
  },
  [a.CLEAR_ALL_FLOCKS](state) {
    const { flocks } = getInitialState();
    const newState = { ...state, flocks };
    return newState;
  },
  [a.JOIN_FLOCK_REQUESTED](state) {
    const newState = {
      ...state,
      joinForm: { inProgress: true, error: null },
    };
    return newState;
  },
  [a.JOIN_FLOCK_FULFILLED](state) {
    const newState = {
      ...state,
      joinForm: { inProgress: false, error: null },
    };
    return newState;
  },
  [a.JOIN_FLOCK_REJECTED](state, action) {
    const { error } = action.payload;
    const newState = {
      ...state,
      joinForm: { inProgress: false, error: error.message },
    };
    return newState;
  },
  [a.ADD_FLOCK_REQUESTED](state) {
    const newState = {
      ...state,
      addForm: { inProgress: true, error: null },
    };
    return newState;
  },
  [a.ADD_FLOCK_FULFILLED](state) {
    const newState = {
      ...state,
      addForm: { inProgress: false, error: null },
    };
    return newState;
  },
  [a.ADD_FLOCK_REJECTED](state, action) {
    const { error } = action.payload;
    const newState = {
      ...state,
      addForm: { inProgress: false, error: error.message },
    };
    return newState;
  },
  [a.DELETE_FLOCK_REQUESTED](state) {
    const newState = {
      ...state,
      deleteFlock: { inProgress: true, error: null },
    };
    return newState;
  },
  [a.DELETE_FLOCK_FULFILLED](state) {
    const newState = {
      ...state,
      deleteFlock: { inProgress: false, error: null },
    };
    return newState;
  },
  [a.DELETE_FLOCK_REJECTED](state, action) {
    const { error } = action.payload;
    const newState = {
      ...state,
      deleteFlock: { inProgress: false, error: error.message },
    };
    return newState;
  },
  [a.DELETE_CHICKEN_REQUESTED](state) {
    const newState = {
      ...state,
      deleteChicken: { inProgress: true, error: null },
    };
    return newState;
  },
  [a.DELETE_CHICKEN_FULFILLED](state) {
    const newState = {
      ...state,
      deleteChicken: { inProgress: false, error: null },
    };
    return newState;
  },
  [a.DELETE_CHICKEN_REJECTED](state, action) {
    const { error } = action.payload;
    const newState = {
      ...state,
      deleteChicken: { inProgress: false, error: error.message },
    };
    return newState;
  },
  [a.SWITCH_FLOCK_REQUESTED](state) {
    const { userSettings } = state;
    const newState = {
      ...state,
      userSettings: {
        ...userSettings,
        initialized: false,
      },
    };
    return newState;
  },
  [a.SWITCH_FLOCK_FULFILLED](state) {
    const { userSettings } = state;
    const newState = {
      ...state,
      userSettings: {
        ...userSettings,
        initialized: true,
      },
    };
    return newState;
  }
};

export default (state = initialState, action = {}) => {
  if (Object.prototype.hasOwnProperty.call(handlers, action.type)) {
    const newState = handlers[action.type](state, action);
    return newState;
  }
  return state;
};
