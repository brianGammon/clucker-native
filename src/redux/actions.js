import firebase from 'react-native-firebase';
import { metaTypes, actionTypes as a } from './constants';

export function firebaseListenRequested(ref, metaType) {
  return {
    type: a.LISTEN_REQUESTED,
    payload: { ref },
    meta: { type: metaType },
  };
}

export function firebaseListenRejected(error, metaType) {
  return {
    type: a.LISTEN_REJECTED,
    payload: { error },
    meta: { type: metaType },
  };
}

export function firebaseListenFulfilled({ key, data }, metaType) {
  return {
    type: a.LISTEN_FULFILLED,
    payload: { key, data },
    meta: { type: metaType },
  };
}

export function firebaseListenChildAdded(key, data, metaType) {
  return {
    type: a.LISTEN_CHILD_ADDED,
    payload: { key, data },
    meta: { type: metaType },
  };
}

export function firebaseListenChildChanged(key, data, metaType) {
  return {
    type: a.LISTEN_CHILD_CHANGED,
    payload: { key, data },
    meta: { type: metaType },
  };
}

export function firebaseListenChildRemoved(key, metaType) {
  return {
    type: a.LISTEN_CHILD_REMOVED,
    payload: { key },
    meta: { type: metaType },
  };
}

export function firebaseUpdateRequested(payload, metaType) {
  return {
    type: a.UPDATE_REQUESTED,
    payload,
    meta: { type: metaType },
  };
}

export function firebaseUpdateRejected(error, metaType) {
  return {
    type: a.UPDATE_REJECTED,
    payload: { error },
    meta: { type: metaType },
  };
}

export function firebaseUpdateFulfilled(metaType) {
  return {
    type: a.UPDATE_FULFILLED,
    payload: {},
    meta: { type: metaType },
  };
}

export function firebaseRemoveRequested(payload, metaType) {
  return {
    type: a.REMOVE_REQUESTED,
    payload,
    meta: { type: metaType },
  };
}

export function firebaseRemoveRejected(error, metaType) {
  return {
    type: a.REMOVE_REJECTED,
    payload: { error },
    meta: { type: metaType },
  };
}

export function firebaseRemoveFulfilled(metaType) {
  return {
    type: a.REMOVE_FULFILLED,
    payload: {},
    meta: { type: metaType },
  };
}

export function firebaseListenRemoved(clearData, metaType) {
  return {
    type: a.LISTEN_REMOVED,
    payload: { clearData },
    meta: { type: metaType },
  };
}

export function firebaseRemoveListenerRequested(clearData, metaType) {
  return {
    type: a.REMOVE_LISTENER_REQUESTED,
    payload: { clearData },
    meta: { type: metaType },
  };
}

export function firebaseRemoveAllListenersRequested() {
  return {
    type: a.REMOVE_ALL_LISTENERS_REQUESTED,
    payload: { clearData: true },
  };
}

export function listenToChickens(flockId) {
  const ref = firebase.database().ref(`chickens/${flockId}`);
  return firebaseListenRequested(ref, metaTypes.chickens);
}

export function listenToEggs(flockId) {
  const ref = firebase.database().ref(`eggs/${flockId}`);
  return firebaseListenRequested(ref, metaTypes.eggs);
}

export function listenToUserSettings(uid) {
  const ref = firebase.database().ref(`userSettings/${uid}`);
  return firebaseListenRequested(ref, metaTypes.userSettings);
}

export function removeUserSettingsListenerRequested() {
  return firebaseRemoveListenerRequested(false, metaTypes.userSettings);
}

// TODO: remove this, no need
export function updateUserSettingsRequested(
  uid,
  displayName,
  currentFlockId,
  flocks,
) {
  return firebaseUpdateRequested(
    {
      uid,
      displayName,
      currentFlockId,
      flocks,
    },
    metaTypes.userSettings,
  );
}

export function removeUserSettingsRequested(uid) {
  return firebaseRemoveRequested({ uid }, metaTypes.userSettings);
}

// TODO: remove refs from actions module
export function getFlock(flockId) {
  const ref = firebase.database().ref(`flocks/${flockId}`);
  return {
    type: a.GET_FLOCK_REQUESTED,
    payload: { ref },
    meta: { type: metaTypes.flocks },
  };
}

export function getFlockRejected(error) {
  return {
    type: a.GET_FLOCK_REJECTED,
    payload: { error },
    meta: { type: metaTypes.flocks },
  };
}

export function getFlockFulfilled(flock) {
  return {
    type: a.GET_FLOCK_FULFILLED,
    payload: flock,
    meta: { type: metaTypes.flocks },
  };
}

export function setInitialUrl(url) {
  return {
    type: a.SET_INITIAL_URL,
    payload: url,
  };
}

export function removeInitialUrl() {
  return {
    type: a.REMOVE_INITIAL_URL,
  };
}
