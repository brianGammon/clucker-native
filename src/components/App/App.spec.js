import React from 'react';
import { shallow } from 'enzyme';
import { Linking } from 'react-native';
import configureStore from 'redux-mock-store';

import ConnectedApp from '.';
import * as actions from '../../redux/actions';
import { appStates } from '../../redux/constants';

describe('App component:', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    appState: appStates.STARTING,
    userSettings: {
      name: 'Test Name',
      currentFlockId: 'Flock11',
      flocks: {
        XYZ123: true,
        XYZ124: true,
      },
    },
    flocks: {
      initialized: true,
    },
  };

  const getStore = state => mockStore(state);

  const getWrapper = store => shallow(<ConnectedApp store={store} />).dive();

  test('Should render Splash when not initialized', () => {
    const wrapper = getWrapper(getStore(initialState));
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().componentWillUnmount();
  });

  test('Should render Root Navigator when initialized', () => {
    const state = { ...initialState, appState: appStates.READY };
    const store = getStore(state);
    const wrapper = getWrapper(store);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should dispatch setInitialUrl when initialUrl is set', (done) => {
    const initialUrl = 'test://url';
    const expectedAction = actions.setInitialUrl(initialUrl);
    Linking.getInitialURL = jest.fn(() => Promise.resolve(initialUrl));
    const store = getStore(initialState);
    const promise = new Promise((resolve) => {
      store.dispatch = jest.fn(() => resolve());
      getWrapper(store);
    });

    promise.then(() => {
      expect(store.dispatch).toBeCalledWith(expectedAction);
      done();
    });
  });

  test('Should not dispatch when no initialUrl is set', (done) => {
    Linking.getInitialURL = jest.fn(() => Promise.resolve(null));
    const store = getStore(initialState);
    const promise = new Promise((resolve) => {
      store.dispatch = jest.fn(() => Promise.resolve());

      getWrapper(store);
      setImmediate(resolve);
    });

    promise.then(() => {
      expect(store.dispatch).not.toBeCalled();
      done();
    });
  });
});
