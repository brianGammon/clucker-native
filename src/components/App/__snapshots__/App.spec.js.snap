// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component: Should render Root Navigator when initialized 1`] = `
<Styled(Root)>
  <Connect(ErrorHandler) />
  <NavigationContainer />
</Styled(Root)>
`;

exports[`App component: Should render Splash when not initialized 1`] = `
<Loading
  message="Welcome to Clucker"
/>
`;
